### PR TITLE
fix(fmgc): flight phase manager improvements

### PR DESCRIPTION
**Pull request by MisterChocker**
## Summary of Changes

- Small improvements to existing code
- Prevents people from flying around in preflight phase.

## Additional context

Discord username (if different from GitHub): DerL30N#3751

## Testing instructions

Perform a flight as per usual and monitor the performance page closely.
While descending re-establish cruise and climb phase at least once by entering a new valid cruise altitude on 1L on the progress page.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js [hunk 1]
```diff
@@ -2638,35 +2638,19 @@ class FMCMainDisplay extends BaseAirliners {
...
+            this.currentFlightPhase === FmgcFlightPhases.DONE
+        ) {
+            return false;
         }
-        if (this.currentFlightPhase === FmgcFlightPhases.GOAROUND) {
+
+        if (this.currentFlightPhase !== FmgcFlightPhases.APPROACH) {
             this.flightPhaseManager.changeFlightPhase(FmgcFlightPhases.APPROACH);
-            Coherent.call("GENERAL_ENG_THROTTLE_MANAGED_MODE_SET", ThrottleMode.AUTO);
-            SimVar.SetSimVarValue("L:A32NX_GOAROUND_PASSED", "bool", 0);
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js:2662]: `GENERAL_ENG_THROTTLE_MANAGED_MODE_SET` and `L:A32NX_GOAROUND_PASSED` are both set on the transition into approach phase. (onFlightPhaseChanged method)

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 10/03/2021
Version of the sim: 1.14.5.0
PR Tested: #3924 
Tier of testing: 2
Changes to observe: Flight phases during a flight

Steps you've made to test the Pull Request:
1. Done a normal short full flight LOWW-EDDM
2. Observed PERF page and A32NX_FMGC_FLIGHT_PHASE LocalVariable
3. Normal CLB to CRZ FL320
4. Step DES FL300
5. Step CLB back to FL320
6. Initiated DES phase
7. Level off FL290 
8. Entered FL300 in PROG
9. CLB to FL300
10. Normal descent, approach and landing
11. Re-established preflight phase at the Gate

Media:

Issues (If Any): - 

Overall Rating: LGTM

Conclusions of the testing: Works as expected. I did not test a go around though.

File: A32NX/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js [hunk 2]
```diff
@@ -142,13 +142,23 @@ class A32NX_FlightPhase_PreFlight {
     }
 
     check(_deltaTime, _fmc) {
+        const ra = Simplane.getAltitudeAboveGround();
+
         return this.takeoffConfirmation.write(
+            Math.round(ra / 100) !== Math.round(Simplane.getAltitude() / 100) && ra > 1.5 ||
```
Comment (beheh) [inline A32NX/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js:148]: Maybe let's add a comment here on exactly why you're doing this, or move it to a helper function. It's pretty obscure if you find this and don't know about the background.

**Review by beheh — APPROVED** [review]

File: A32NX/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js (hunk 2 above)
Comment (MisterChocker) [reply A32NX/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js:148]: absolutely.

Comment (MisterChocker) [inline A32NX/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js:148]:
```suggestion
        return this.takeoffConfirmation.write(
            // we try to detect a false lift off (during terrain loading) from a true liftoff e.g. during takeoff. (temporary solution only)
            Math.round(ra / 100) !== Math.round(Simplane.getAltitude() / 100) && ra > 1.5 ||
```

**Comment by sidnov** [comment]
Category: QA Tester
Name: sidnov#8337
Date of testing: 18.3.2021
Version of the sim: 1.14.5.0
PR Tested: #3924
Tier of testing: 2

### Changes to observe: flight phase manager during descent and re-established cruise

### Testing technique: Tryout

### Steps made to test the Pull Request:

1. Spawned C/D, prepped for a routine IFR flight
2. Departed VOMM, climbed to cruise FL380
3. Descended to FL360 for 10 mins
4. Climbed back to FL380, PROG page updated as expected.
5. Descended to FL320, checked page updating.
6. 120nm from DEST, set descent to FL190. PERF mode entered DES.
7. Stopped midway around FL260, reinitiated CRZ to FL280 through the PROG page.
8. Observed necessary PERF pages visible. Reinitiated descent in a few miles.
9. Approach and Landing as normally observed.

### Notes:
VOMM DOLKI Q8 AGELA VABB
Weather: default
Navdata: Navigraph Beta MSFS

### Issues (If Any): None

### Overall Rating: Good

### Conclusions of the testing: 
The flight phases update as expected.

**Comment by TheSquirtingPantin** [comment]
Category: QA Trial
Name: Discord User Rémi#8871
Date of testing: 19/03/2021
Version of the sim: V1.14.5.0
PR Tested: #3924
Tier of testing: 2
Changes to observe: Change of flight mode on PERF page

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Performed a flight KPDX/KDEN at FL350
2. Inside the 200nm to destination, descended to FL330
3. Levelled off at FL330
4. New CRZ ALT entered in PROG page 1L (FL350) 
5. Observed changes on PERF page

Flight notes (When done a full flight):
KPDX/10R WHAMY4 IMB DCT BOI DCT PIH DCT OCS DCT MJANE FLATI2 KDEN/26

Media:N/A

Issues (If Any): No issues. PERF page logic is as expected. 

Overall Rating: Good
Conclusions of the testing: LGTM

**Pull request merged**